command.go

```diff
@@ -207,29 +221,39 @@ func (cmd *CommandFunc) Call(args, env []string) (int, error) {
 
 	var params []reflect.Value
 
-	if t := cmd.function.Type(); t.NumIn() > 0 {
-		// Configuration options are decoded into the first function parameter.
-		v := reflect.New(t.In(0)).Elem()
-		if err := cmd.options.decode(v, options); err != nil {
-			return 1, err
-		}
-		params = append(params, v)
+	x := 0
 
+	if cmd.context {
+		params = append(params, reflect.ValueOf(ctx))
+		x++
+	}
```

Comment:
Can we add this?
```go
} else if ctx != context.TODO() { // or context.Background() depending on which one is more appropriate
    panic("mixing usage of context aware and unaware commands")
}
```
That way we prevent ignoring non-background contexts that would be passed to commands that are not context-aware.